codesign/lib/src/codesign_context.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+import 'package:file/local.dart';
+import 'package:process/process.dart';
+import 'package:codesign/codesign.dart';
+import 'package:file/file.dart';
+
+class CodesignContext {
+  CodesignContext({
+    required this.codesignCertName,
+    required this.codesignPrimaryBundleId,
+    required this.codesignUserName,
+    required this.appSpecificPassword,
+    required this.codesignAppstoreId,
+    required this.codesignTeamId,
+    required this.codesignFilepath,
+    required this.commitHash,
+    this.production = false,
+  });
+
+  final String codesignCertName;
+  final String codesignPrimaryBundleId;
+  final String codesignUserName;
+  final String appSpecificPassword;
+  final String codesignAppstoreId;
+  final String codesignTeamId;
+  final String codesignFilepath;
+  final String commitHash;
+  final bool production;
+  Directory? tempDir;
+  FileCodesignVisitor? codesignVisitor;
+
+  ProcessManager processManager = LocalProcessManager();
+  FileSystem fileSystem = LocalFileSystem();
+  Stdio stdio = VerboseStdio(
+    stdout: stdout,
+    stderr: stderr,
+    stdin: stdin,
+  );
+
+  bool checkXcodeVersion() {
+    bool isNotaryTool = true;
+    print('checking Xcode version...');
+    final ProcessResult result = processManager.runSync(
+      <String>[
+        'xcodebuild',
+        '-version',
+      ],
+    );
+    final List<String> outArray = (result.stdout as String).split('\n');
+    final int xcodeVersion = int.parse(outArray[0].split(' ')[1].split('.')[0]);
+    if (xcodeVersion <= 12) {
```

Comment:
I think we're on Xcode 13+ everywhere, if you want to remove all the `isNotaryTool` fallback logic.